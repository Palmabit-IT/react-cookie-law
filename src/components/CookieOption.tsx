import React from 'react';

interface Props {
  checked?: boolean
  onChange?: (checked: boolean) => void
  id?: string
  text?: string
  styles?: Record<string, React.CSSProperties>
  disabled?: boolean
}

interface State {
  checked: boolean
}

class CookieOption extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);

    const { checked = false } = props;
    this.state = { checked };
    this.handleOnChange = this.handleOnChange.bind(this);
  }

  handleOnChange() {
    const { onChange = () => undefined } = this.props;
    const { checked } = this.state;
    const newValue = !checked;

    this.setState({ checked: newValue });
    onChange(newValue);
  }

  render() {
    const {
      id = '',
      text = '',
      styles = {},
      disabled = false,
    } = this.props;

    const isDisabled = disabled ? { disabled } : {};
    const { checked } = this.state;

    const { optionWrapperStyle, optionLabelStyle, checkboxStyle } = styles;

    return (
      <div
        className="react-cookie-law-option-wrapper"
        style={optionWrapperStyle}
      >
        <input
          type="checkbox"
          id={id}
          className="react-cookie-law-option-checkbox"
          style={checkboxStyle}
          checked={checked}
          onChange={this.handleOnChange}
          {...isDisabled}
        />
        <label htmlFor={id} style={optionLabelStyle}>
          {text}
        </label>
      </div>
    );
  }
}

export default CookieOption;
