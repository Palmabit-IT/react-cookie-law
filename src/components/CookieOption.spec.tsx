import React from 'react';
import { mount, shallow } from 'enzyme';
import CookieOption from './CookieOption';

describe('CookieOption component', () => {
  test('should be rendered', () => {
    const props = {
      id: 'check-required-cookies',
      text: 'Necessary',
      disabled: false,
      checked: false,
    };

    const component = shallow(
      <CookieOption {...props} />,
    );

    expect(component).toMatchSnapshot();
  });

  test('should be rendered with elements', () => {
    const props = {
      id: 'check-cookies',
      text: 'Custom label text',
      disabled: false,
      checked: false,
    };

    const component = mount(
      <CookieOption {...props} />,
    );

    expect(component.find('input').prop('id')).toBe('check-cookies');
    expect(component.find('.react-cookie-law-option-checkbox').props().disabled).toBeFalsy();
    expect(component.find('.react-cookie-law-option-checkbox').props().checked).toBeFalsy();
    expect(component.find('label').text()).toBe('Custom label text');
  });

  test('should be rendered checked element', () => {
    const props = {
      id: 'check-cookies',
      text: 'Custom label text',
      disabled: true,
      checked: true,
    };

    const component = mount(
      <CookieOption {...props} />,
    );

    expect(component.find('.react-cookie-law-option-checkbox').props().disabled).toBeTruthy();
    expect(component.find('.react-cookie-law-option-checkbox').props().checked).toBeTruthy();
  });

  test('should call onChange callback', () => {
    const props = {
      onChange: jest.fn(),
    };

    const component = mount<CookieOption>(
      <CookieOption {...props} />,
    );

    component.find('.react-cookie-law-option-checkbox').simulate('change');

    expect(props.onChange).toHaveBeenCalledTimes(1);
    expect(component.instance().state.checked).toBeTruthy();
  });
});
