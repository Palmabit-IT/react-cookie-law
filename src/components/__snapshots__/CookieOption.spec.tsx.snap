// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CookieOption component should be rendered 1`] = `
<div
  className="react-cookie-law-option-wrapper"
>
  <input
    checked={false}
    className="react-cookie-law-option-checkbox"
    id="check-required-cookies"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    htmlFor="check-required-cookies"
  >
    Necessary
  </label>
</div>
`;
